src/engine/guidance/post_processing.cpp

```diff
@@ -88,13 +88,66 @@ void processRoundaboutExits(const RouteStepIterator begin, const RouteStepIterat
         return;
     }
 
-    const auto passes_exit_or_leaves_roundabout = [](auto const &step) {
-        return staysOnRoundabout(step.maneuver.instruction) ||
-               leavesRoundabout(step.maneuver.instruction);
-    };
+    const auto exit = std::accumulate(begin, end, 0, [](const long a, const auto &step) {
+        if (staysOnRoundabout(step.maneuver.instruction))
+        {
+            // Count all the possible exits from from the step, minus one, which is the
+            // "continue on the roundabout" exit
+            const auto step_intersection = step.intersections.front();
+            const auto num_exits_from_step = std::count_if(std::begin(step_intersection.entry),
```

Comment:
This logic doesn't work for throughabouts. If a road (bidirectional) passes through a roundabout, the exit will be counted twice.

Reply:
Also, this would need to ignore exits that cannot be entered (exit against roundabout direction)